crates/resolver/src/config.rs

```diff
@@ -731,6 +731,25 @@ impl Default for LookupIpStrategy {
     }
 }
 
+/// The strategy for establishing the query order of name servers in a pool.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "serde-config", derive(Serialize, Deserialize))]
+pub enum ServerOrderingStrategy {
+    /// Servers are ordered based on collected query statistics. The ordering
+    /// may vary over time.
+    Default,
+    /// The order provided to the resolver is used. The ordering does not vary
+    /// over time.
+    Strict,
```

Comment:
Nit: IMO "strict" isn't great at explaining the intended semantics, maybe something like UserProvidedOrder (overly verbose) or some such? Similarly "default" is kind of tautological, could be LatencyWeighted?

Reply:
Good points. The "Strict" naming was inspired from Dnsmasq, but I agree that UserProvidedOrder makes more sense. Additionally, I've updated "Default" to "QueryStatistics" as the current implementation isn't influenced by latency. I'd be happy to adopt the suggested naming if you'd prefer though since we do plan to have it be based on latency soon.

Reply:
QueryStatistics is fine too, we can consider renaming it once you land the new strategy.